model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt

```diff
@@ -79,14 +82,20 @@ class ModelClientV2(
     private var clientId: Int = 0
     private var idGenerator: IIdGenerator = IdGeneratorDummy()
     private var serverProvidedUserId: String? = null
-    private val kvStore = MapBasedStore()
-    val store = ObjectStoreCache(kvStore) // TODO the store will accumulate garbage
+
+    // TODO the store will accumulate garbage
+    private val storeForRepository: MutableMap<RepositoryId, IDeserializingKeyValueStore> = HashMap()
 
     suspend fun init() {
         updateClientId()
         updateUserId()
     }
 
+    private fun getStore(repository: RepositoryId) = runSynchronized(storeForRepository) { storeForRepository.getOrPut(repository) { ObjectStoreCache(MapBasedStore()) } }
+    private fun getRepository(store: IDeserializingKeyValueStore): RepositoryId {
+        return storeForRepository.asSequence().first { it.value == store }.key
+    }
```

Comment:
How often will this be called? Due to the inverse operation, this cannot make use of an efficient lookup.